remotespark/livyclientlib/livysession.py

```
@@ -65,21 +68,26 @@ def start(self, create_sql_context=True):
         """Start the session against actual livy server."""
         self._spark_events.emit_session_creation_start_event(self.guid, self.kind)
 
-        r = self._http_client.post_session(self.properties)
-        self.id = r["id"]
-        self.status = str(r["state"])
-
-        self.ipython_display.writeln("Creating SparkContext as 'sc'")
-        # We wait for livy_session_startup_timeout_seconds() for the session to start up.
         try:
-            self.wait_for_idle(conf.livy_session_startup_timeout_seconds())
-        except LivyClientTimeoutError:
-            raise LivyClientTimeoutError("Session {} did not start up in {} seconds."\
-                                         .format(self.id, conf.livy_session_startup_timeout_seconds()))
-
-        if create_sql_context:
-            self.create_sql_context()
-        self._spark_events.emit_session_creation_end_event(self.guid, self.kind, self.id, self.status)
+            r = self._http_client.post_session(self.properties)
+            self.id = r["id"]
+            self.status = str(r["state"])
+
+            self.ipython_display.writeln("Creating SparkContext as 'sc'")
+            # We wait for livy_session_startup_timeout_seconds() for the session to start up.
+            try:
+                self.wait_for_idle(conf.livy_session_startup_timeout_seconds())
+            except LivyClientTimeoutError:
+                raise LivyClientTimeoutError("Session {} did not start up in {} seconds."\
+                                             .format(self.id, conf.livy_session_startup_timeout_seconds()))
+
+            if create_sql_context:
+                self.create_sql_context()
+        except Exception as e:
+            self._spark_events.emit_session_creation_end_event(self.guid, self.kind, self.id, self.status,
+                                                               False, e.__class__.__name__, str(e))
```

You need to re-raise the exception afterward.